flytekit/core/reference_entity.py

```diff
@@ -75,23 +75,22 @@ def __init__(
         ):
             raise Exception("Must be one of task, workflow, or launch plan")
         self._reference = reference
-        self._interface = Interface(inputs=inputs, outputs=outputs)
-        self._typed_interface = transform_interface_to_typed_interface(self._interface)
+        self.reset_interface(inputs, outputs)
 
     def execute(self, **kwargs) -> Any:
         raise Exception("Remote reference entities cannot be run locally. You must mock this out.")
 
-    def reset_interface(self, inputs: Dict[str, Type], outputs: Dict[str, Type]):
-        self._interface = Interface(inputs=inputs, outputs=outputs)
-        self._typed_interface = transform_interface_to_typed_interface(self._interface)
+    def reset_interface(self, inputs: Optional[Dict[str, Type]], outputs: Dict[str, Type]):
```

Comment:
@wild-endeavor why do we have a mutation on this API? I actually do not like mutations on such an api?

Reply:
I think I forgot to delete this.  Shouldn't be used.  @cosmicBboy can you delete?